Frameworks/Foundation/NSHTTPURLResponse.mm

```diff
@@ -54,19 +72,7 @@ - (id)initWithURL:(NSURL*)url statusCode:(NSInteger)statusCode HTTPVersion:(NSSt
     NSMutableDictionary* allHeaderFields = nil;
 
     if (headerFields != nil) {
-        CFDictionaryKeyCallBacks caseInsensitiveKeyChecker = kCFTypeDictionaryKeyCallBacks;
-
-        caseInsensitiveKeyChecker.equal = _CFHTTPHeaderEqual;
-        caseInsensitiveKeyChecker.hash = _CFHTTPHeaderHash;
-
-        allHeaderFields = (NSMutableDictionary*)
-            CFDictionaryCreateMutable(NULL, [headerFields count], &caseInsensitiveKeyChecker, &kCFTypeDictionaryValueCallBacks);
-
-        //  Case insensitive dictionary
-        for (id key in [headerFields allKeys]) {
-            id val = [headerFields objectForKey:key];
-            [allHeaderFields setObject:val forKey:key];
-        }
+		allHeaderFields = _constructCaseInsensitiveDictionary(headerFields);
```

Comment:
below, too

---
In reply to: [72351397](https://github.com/Microsoft/WinObjC/pull/673#discussion_r72351397) [](ancestors = 72351397)